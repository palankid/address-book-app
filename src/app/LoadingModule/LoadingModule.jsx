import React from 'react';

import { Icon } from 'semantic-ui-react';

/**
 * Fallback component to display when a component for a specific
 * route is not loaded and therefore can't be displayed yet
 * @returns {Object<Node>} React node
 */
const LoadingModule = () => {
  return (
    <div className="loading-module">
      <Icon
        loading
        name="spinner"
        size="huge"
      />
      <span className="loading-module__text">
        Loading module
      </span>
    </div>
  );
};

export default LoadingModule;
