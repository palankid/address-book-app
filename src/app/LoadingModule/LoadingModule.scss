.loading-module {
  display: flex;
  align-items: center;
  justify-content: center;
  height: 100%;

  &__text {
    font-size: 1.5rem;
    padding: 10px;
    color: #FFFFFF;
    background-color: forestgreen;
  }
}
