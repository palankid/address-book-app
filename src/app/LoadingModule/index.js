import LoadingModule from './LoadingModule';
import './LoadingModule.scss';

export default LoadingModule;
