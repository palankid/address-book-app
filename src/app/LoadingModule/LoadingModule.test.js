import React from 'react';
import { shallow } from 'enzyme';

import LoadingModule from './LoadingModule';

describe('LoadingModule', () => {
  it('should test that component is not changed unintentionally', () => {
    const wrapper = shallow(<LoadingModule />);

    expect(wrapper).toMatchSnapshot();
  });
});
