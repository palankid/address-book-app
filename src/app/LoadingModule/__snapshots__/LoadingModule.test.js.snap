// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingModule should test that component is not changed unintentionally 1`] = `
<div
  className="loading-module"
>
  <Icon
    as="i"
    loading={true}
    name="spinner"
    size="huge"
  />
  <span
    className="loading-module__text"
  >
    Loading module
  </span>
</div>
`;
